### ISSUE TITLE
No such module 'EXDevLauncher'

### ISSUE THREAD

**Issue by adamsadiq7** [expo/expo#18150]
### Summary

When running an archive on Xcode, during the last stages the "No such module 'EXDevLauncher'" error is being displayed. Additionally I get "Command EmitSwiftModule failed with a nonzero exit code". I've followed all the steps in the issue here: https://github.com/expo/expo/issues/17246 with no avail.

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

bare

### What platform(s) does this occur on?

iOS

### SDK Version (managed workflow only)

SDK45

### Environment

expo-env-info 1.0.4 environment info:
    System:
      OS: macOS 12.4
      Shell: 5.8.1 - /bin/zsh
    Binaries:
      Node: 16.15.0 - /usr/local/bin/node
      npm: 8.5.5 - /usr/local/bin/npm
      Watchman: 2022.05.16.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.11.3 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 21.4, iOS 15.5, macOS 12.3, tvOS 15.4, watchOS 8.5
    IDEs:
      Android Studio: 2021.2 AI-212.5712.43.2112.8512546
      Xcode: 13.4/13F17a - /usr/bin/xcodebuild
    npmPackages:
      expo: ^45.0.0 => 45.0.4 
      react: 17.0.2 => 17.0.2 
      react-dom: 17.0.2 => 17.0.2 
      react-native: 0.68.2 => 0.68.2 
      react-native-web: 0.17.7 => 0.17.7 
    npmGlobalPackages:
      eas-cli: 0.54.1
      expo-cli: 5.4.6
    Expo Workflow: bare

### Reproducible demo

Archive from Xcode

**Comment by adamsadiq7** [comment]
Sorry it took so long, manic week! It's just below, I think it matches what you provided

```
/**
 * Automatically generated by expo-modules-autolinking.
 *
 * This autogenerated class provides a list of classes of native Expo modules,
 * but only these that are written in Swift and use the new API for creating Expo modules.
 */

import ExpoModulesCore
import EASClient
import ExpoHaptics
import ExpoImagePicker
import ExpoKeepAwake
import ExpoSystemUI
import EXUpdates
#if EXPO_CONFIGURATION_DEBUG
import EXDevLauncher
import EXDevMenu
#endif

@objc(ExpoModulesProvider)
public class ExpoModulesProvider: ModulesProvider {
  public override func getModuleClasses() -> [AnyModule.Type] {
    return [
      EASClientModule.self,
      HapticsModule.self,
      ImagePickerModule.self,
      KeepAwakeModule.self,
      ExpoSystemUIModule.self
    ]
  }

  public override func getAppDelegateSubscribers() -> [ExpoAppDelegateSubscriber.Type] {
    #if EXPO_CONFIGURATION_DEBUG
    return [
      ExpoUpdatesAppDelegateSubscriber.self,
      ExpoDevLauncherAppDelegateSubscriber.self
    ]
    #else
    return [
      ExpoUpdatesAppDelegateSubscriber.self
    ]
    #endif
  }

  public override func getReactDelegateHandlers() -> [ExpoReactDelegateHandlerTupleType] {
    #if EXPO_CONFIGURATION_DEBUG
    return [
      (packageName: "expo-updates", handler: ExpoUpdatesReactDelegateHandler.self),
      (packageName: "expo-dev-launcher", handler: ExpoDevLauncherReactDelegateHandler.self),
      (packageName: "expo-dev-menu", handler: ExpoDevMenuReactDelegateHandler.self)
    ]
    #else
    return [
      (packageName: "expo-updates", handler: ExpoUpdatesReactDelegateHandler.self)
    ]
    #endif
  }
}

```

**Comment by ajsmth** [comment]
Yes that looks good to me - the only thing I can think of is that you have a local `DEBUG` environment variable that might be overriding what is expected here - if you can produce a minimal reproduction I can have a closer look!

**Comment by fullstack17** [comment]
@adamsadiq7 I'm still having got this issue although my current code is same as the above. expo version is 45 and it's bare expo.
<img width="925" alt="Screenshot 2022-09-21 at 08 36 03" src="https://user-images.githubusercontent.com/44271282/191422610-b462a136-c4b1-4f85-ad55-001479c63936.png">

**Comment by ste7en** [comment]
I'm having the same issue, anyone succeeded in solving this?

**Comment by wsanada** [comment]
Same here too. I've been stuck for days.

It's a new project
...
"expo": "^46.0.16",
"expo-updates": "~0.14.6",
"react": "18.0.0",
"react-native": "0.69.6",
...

**Comment by ste7en** [comment]
I solved setting the "Build active architectures only" to NO in the Pod project

**Comment by wsanada** [comment]
> I solved setting the "Build active architectures only" to NO in the Pod project

It didn't work for me :(

![image](https://user-images.githubusercontent.com/41298785/201236389-0d6ef639-7aa7-4a28-b5a1-e28fe13e5473.png)

**Comment by wilburlua910** [comment]
i'm having the same issue, although Release configuration was specified when archiving, debug flag was still set to true. I solved this temporarily by uninstalling the expo-dev-client package, and i could finally use eas build to generate the release binary build.

**Comment by wsanada** [comment]
I solved it by removing "-D EXPO_CONFIGURATION_DEBUG" statement from TARGETS > Build Settings > Other Swift Flags > Release in my XCode project.

So it goes like this:
![image](https://user-images.githubusercontent.com/41298785/203792664-90ee429d-f670-46c5-b77e-114a5028747e.png)

I hope it helps

**Comment by CesarCanoff** [comment]
**Same error here:**

expo: "~48.0.15",
expo-updates: "^0.18.17",
react: "18.2.0",
react-native: "0.71.14"

**Comment by jeetvani** [comment]
> I solved it by removing "-D EXPO_CONFIGURATION_DEBUG" statement from TARGETS > Build Settings > Other Swift Flags > Release in my XCode project.
> 
> So it goes like this: ![image](https://user-images.githubusercontent.com/41298785/203792664-90ee429d-f670-46c5-b77e-114a5028747e.png)
> 
> I hope it helps

Amaing Woked for me too .. !
